### ISSUE TITLE
FP S3863: ignore import statements for the sake of side-effects

### ISSUE THREAD

**Issue by yassin-kammoun-sonarsource** [SonarSource/SonarJS#2735]
```javascript
import "abc";
import { X } from "abc";
```

The import statement without specifiers is likely to have side-effects necessary for using the library.

Reported in https://community.sonarsource.com/t/import-for-side-effects-is-flagged-as-duplicated/30953

**Comment by thw0rted** [comment]
LGTM. When I made my Discourse post, TS 3.8 had only introduced the `import type` syntax a few months prior, and the library I was using hadn't updated their docs yet.

Actually, hey, there's an idea -- can we get a rule that flags `import {X}` if `X` is never used in a value position?  I've tried to convert to `import type` wherever possible but sometimes I miss them, and it really doesn't help that VSCode automatic import never uses `import type`.  (NB: this is hard to get right because sometimes a type used in a type-only position still impacts runtime via decorator metadata, like with Angular DI.)

**Comment by vilchik-elena** [comment]
@thw0rted we were thinking about it in this ticket https://github.com/SonarSource/eslint-plugin-sonarjs/issues/266, rule idea is that imported dev dependency is used in production not only for types. But indeed the way you see it (any import if it's used only for types should be marked so) has a more narrow scope. I created a ticket https://github.com/SonarSource/SonarJS/issues/2759
